File: plots.py
# %%
import pandas as pd
import json
import os
import matplotlib.pyplot as plt
import seaborn as sns

# %%

project="SC"
genes=["RORB","LAMP5"]

umap_embeddings_file = os.path.join(project, 'umap_embeddings_with_meta.csv')
data_df = pd.read_csv(umap_embeddings_file, index_col=0, header=0)
## Cell,UMAP_1,UMAP_2,sample_id,case,sex,age,seurat_clusters,MajorCellTypes,CellSubtypes

# %%
for gene in genes:
    gene_expr_file = os.path.join(project, "gene_jsons",gene+".json")
    with open(gene_expr_file, 'r') as f:
        cell_expr = json.load(f)
        data_df[gene] = data_df.index.map(cell_expr).fillna(0)
data_df = data_df.loc[:, ["UMAP_1", "UMAP_2","MajorCellTypes"]+genes]

# %%
sns.violinplot(data=data_df, x="MajorCellTypes", y="RORB", palette="Set1")
plt.xticks(rotation=90)
plt.title("RORB - All (500k)")
plt.savefig("RORB_violin.png")

sns.violinplot(data=data_df, x="MajorCellTypes", y="LAMP5", palette="Set1")
plt.xticks(rotation=90)
plt.title("LAMP5 - All (500k)")
plt.savefig("LAMP5_violin.png")


# %%
# Create scatter plot
plt.figure(figsize=(8, 6))
sns.scatterplot(data=data_df, x="UMAP_1", y="UMAP_2", hue="MajorCellTypes", palette="Set1", s=3)
plt.title("500k")
plt.legend(title="MajorCellTypes", bbox_to_anchor=(1.05, 1), loc='upper left')
plt.savefig("MajorCellTypes_Scatter.png")

# Create scatter plot
plt.figure(figsize=(8, 6))
sns.scatterplot(data=data_df, x="UMAP_1", y="UMAP_2", hue="RORB", s=3)
plt.title("500k")
plt.legend(title="RORB", bbox_to_anchor=(1.05, 1), loc='upper left')
plt.savefig("RORB_Scatter.png")

# Create scatter plot
plt.figure(figsize=(8, 6))
sns.scatterplot(data=data_df, x="UMAP_1", y="UMAP_2", hue="LAMP5", s=3)
plt.title("500k")
plt.legend(title="LAMP5", bbox_to_anchor=(1.05, 1), loc='upper left')
plt.savefig("LAMP5_Scatter.png")
# %%

File: plots.r
library(ggplot2)
library(jsonlite)
library(data.table)

project <- "SC"
genes <- c("RORB", "LAMP5")

# Read UMAP embeddings
umap_file <- file.path(project, "umap_embeddings_with_meta_30k.csv")
data_df <- fread(umap_file)
setnames(data_df, names(data_df)[1], "Cell")  # Set first column as 'Cell'

# Add gene expression data
for(gene in genes) {
  gene_file <- file.path(project, "gene_jsons", paste0(gene, ".json"))
  gene_data <- fromJSON(gene_file)
  
  # Convert JSON to data.frame
  gene_df <- stack(gene_data)
  colnames(gene_df) <- c(gene, "Cell")
  
  # Merge with main data
  data_df <- merge(data_df, gene_df, by = "Cell", all.x = TRUE)
  
  # Replace NA with 0
  data_df[is.na(get(gene)), (gene) := 0]
}

# Create violin plots
create_violin <- function(gene) {
  ggplot(data_df, aes(x = MajorCellTypes, y = .data[[gene]], fill=MajorCellTypes)) +
    geom_violin(trim = FALSE, scale = "width") +
    # geom_boxplot(width=0.1) +
    # geom_jitter(width = 0.1, alpha = 0.5) +
    labs(title = paste(gene, "Expression - 30k"), x = "Major Cell Types", y = "Expression Level") +
    theme_minimal() +
    theme(axis.text.x = element_text(angle = 45, hjust = 1))
}

# Generate and save plots
for(gene in genes) {
  ggsave(
    filename = paste0(gene, "_violin_R.pdf"),
    plot = create_violin(gene),
    device = "pdf",
    width = 8,
    height = 6
  )
}

# Create scatter plots
create_scatter_continuous <- function(feature) {
  ggplot(data_df, aes(x = UMAP_1, y = UMAP_2, color = .data[[feature]])) +
    geom_point(size = 0.6) +
    scale_color_gradient(low = "gray", high = "red") +  # Color gradient from gray to red
    labs(title = "30k", x = "UMAP_1", y = "UMAP_2") +
    theme_minimal()
}

create_scatter<- function(feature) {
  ggplot(data_df, aes(x = UMAP_1, y = UMAP_2, color = .data[[feature]])) +
    geom_point(size = 0.6) +
    labs(title = "30k", x = "UMAP_1", y = "UMAP_1") +
    theme_minimal()
}

# Generate and save plots
for(gene in genes){
  print(gene)
  ggsave(
    filename = paste0(gene, "_scatter_R.pdf"),
    plot = create_scatter_continuous(gene),
    device = "pdf",
    width = 8,
    height = 6
  )
}


ggsave(
    filename = paste0("MajorCellTypes", "_scatter_R.pdf"),
    plot = create_scatter("MajorCellTypes"),
    device = "pdf",
    width = 8,
    height = 6
    )
